examples/traffic-split.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: core.hydra.io/v1alpha1
+kind: ApplicationConfiguration
+metadata:
+  name: example-multi-server
+spec:
+  components:
+    - name: nginx-singleton-v1
+      instanceName: multi-singleton-1
+      traits:
+        - name: traffic-split
+          parameterValues:
+            - name: splitterName
+              value: example-traffic-splitter # <-- Note, same as below
+            - name: weight
+              value: 30
+    - name: nginx-singleton-v1
+      instanceName: multi-singleton-2
```

Comment:
Two component with same version can cause misleading, do you mean two component versions?

But if we put two versions of one component together with different instanceName in the spec, that would not be very elegant solution for traffic split trait .

Traffic split usually used for component rollout or two different components.

For component rollout here, I suggest we could just replace component in the spec. Let the runtime(Rudr) record these changes. For example, let the componentInstance to remember what's his parent spec.

We internally have discussed traffic split many times. In some cases, we really need to record two versions of one component  in the spec, for example multi-clusters. In this case, OAM is like a kind of protocol to transfer our demand to under clusters, we have no elsewhere to record but in the spec.